pkg/apis/scheduling/v1alpha1/types_location.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Location represents a scheduling target for instances of a given type. A location
+// is a projection and aggregation of instances referenced in a LocationDomain, used
+// to express scheduling choices by a user. A location can subsume a number of instances.
+//
+// The location is chosen by the user (in the future) through a Placement object, while
+// the instance is chosen by the scheduler depending on considerations like load
+// or available resources, or further node selectors specified by the user.
+//
+// +crd
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories=kcp
+// +kubebuilder:printcolumn:name="Type",type=string,JSONPath=`.spec.type`,description="Type of the workspace"
+// +kubebuilder:printcolumn:name="Available",type=string,JSONPath=`.status.availableInstances`,description="Available instances in this location"
+// +kubebuilder:printcolumn:name="Instances",type=string,JSONPath=`.status.instances`,description="Instances in this location"
+// +kubebuilder:printcolumn:name="Labels",type=string,JSONPath=`.metadata.annotation."scheduling.kcp.dev/labels"`,description="The common labels of this location"
+type Location struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec LocationSpec `json:"spec,omitempty"`
+
+	// +optional
+	Status LocationStatus `json:"status,omitempty"`
+}
+
+// LocationSpec holds the desired state of the Location.
+type LocationSpec struct {
+	LocationSpecBase `json:",inline"`
+
+	// domain is the LocationDomain this location belongs to.
+	// +required
+	// +kubebuilder:Required
+	Domain LocationDomainReference `json:"domain"`
+
+	// type defines which class of objects this location can schedule. A typical
+	// type is "Workloads".
+	//
+	// +required
+	// +kubebuilder:Required
+	// +kubebuilder:validation:MinLength=1
+	Type string `json:"type,omitempty"`
+}
+
+type LocationSpecBase struct {
+	// description is a human-readable description of the location.
+	//
+	// +optional
+	Description string `json:"description,omitempty"`
+
+	// labels is a set of labels that all instances in this location have in common.
+	//
```

Comment:
good question. We could store them as labels too. Was thinking this is more visible. Was also thinking to return a string represenation for table view in kubectl. So that would bring the visibility we are after.

Reply:
Moving them just into the `LocationDomain` and we will set them as normal ObjectMeta.Labels.